docs/40_smart-contract-guides/80_linking-custom-permission.md

```diff
@@ -2,72 +2,585 @@
+EOSIO permissions control what an account is authorized to execute on an EOSIO blockchain. EOSIO permissions are powerful, flexible and customizable. This tutorial explains custom permissions and shows you how to use them. In the tutorial you will create a custom permission; then link the permission to an action.
+## Before You Begin
 
-This tutorial illustrates the creation of a custom permission, and subsequently, how to link the permission to an action. Upon completion of the steps, the contract's action will be prohibited from executing unless the authorization of the newly linked permission is provided. This allows you to have greater granularity of control over an account and its various actions.
+This tutorial requires the following:
 
-With great power comes great responsibility. This functionality poses some challenges to the security of your contract and its users. Ensure you understand the concepts and steps prior to putting them to use.
+* Access to a running blockchain. Click on this link for instructions on [running a blockchain](01_before-you-begin/10_running-a-blockchain.md)
+* An EOSIO account and access to the account keys. Click on this link for information on [Accounts and Permissions](01_before-you-begin/20_accounts-and-permissions.md)
+* Some knowledge of smart contracts, we recommend following the [hello world tutorial](../30_getting-started-guide/25_hello-world.md) tutorial
 
-[[info |Parent permission ]]
-| When you create a custom permission, the permission will always be created under a parent permission.
+## What is a Custom Permission
 
-If you have the authority of a parent permission which a custom permission was created under, you can always execute an action which requires that custom permission.
+When an account is created two permissions are created by default; `owner` and `active`. You can create a new permission, a custom permission, as a child permission of `owner`, `active` or another custom permission. Custom permissions require a public and private key pair. These custom permissions may be linked to an action of a smart contract and specify the permission required to execute that action. This permission system enables accounts to have flexible and highly granular control over accounts and smart contract actions.
 
-## Step 1. Create a Custom Permission
-Firstly, let's create a new permission level on the `alice` account:
+[[info]]
+| The permission `eosio.code` is a special permission which allows smart contracts to call inline actions. Smart contracts do not have access to account keys and the EOSIO blockchain performs an authorization check when an inline action is called. Adding `eosio.code` to an account permission gives explicit permission to execute code. For information on how to add the `eosio.code` see [cleos set account permission](https://developers.eos.io/manuals/eos/latest/cleos/command-reference/set/set-account-permission.md)
+
+## Why use a Custom Permission
```

Comment:
Title question and make it more formal:
Why use a Custom Permission -> Why us**ing** a Custom Permission?